From 6137d263f3805c72781ea74b813b0e3754f5fc34 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Tue, 17 Mar 2020 16:20:52 +0100
Subject: [PATCH 08/16] ucm: fix uc_mgr_scan_master_configs()

Skip the lookup, if the top level filename does not exist.

Fixes: https://github.com/alsa-project/alsa-ucm-conf/issues/16

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 src/ucm/parser.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/ucm/parser.c b/src/ucm/parser.c
index 8d6eea31f8e7..0001e5953707 100644
--- a/src/ucm/parser.c
+++ b/src/ucm/parser.c
@@ -1969,6 +1969,9 @@ int uc_mgr_scan_master_configs(const char **_list[])
 
 		configuration_filename2(filename, sizeof(filename), 2,
 					d_name, d_name, ".conf");
+		if (eaccess(filename, R_OK))
+			continue;
+
 		err = uc_mgr_config_load(2, filename, &cfg);
 		if (err < 0)
 			goto __err;
-- 
2.16.4

